src/plugins/clips-executive/clips/domain.clp

```diff
@@ -216,263 +636,187 @@
+                    (retract ?grounding)
+  )
 )
 
-
-(deffunction domain-is-precond-negative
-  "Check if a non-atomic precondition is negative by checking all its parents
-   and counting the number of negations. If the number is odd, the precondition
+(deffunction domain-is-formula-negative
+  "Check if a formula is negative by checking all its parents
+   and counting the number of negations. If the number is odd, the formula
    is negative, otherwise it's positive."
-  (?precond-name)
+  (?precond-id)
   (do-for-fact
-    ((?precond domain-precondition))
-    (eq ?precond:name ?precond-name)
+    ((?precond pddl-formula))
+    (eq ?precond:id ?precond-id)
     (if (any-factp ((?op domain-operator)) (eq ?op:name ?precond:part-of)) then
       return (eq ?precond:type negation)
     )
-    (bind ?parent-is-negative (domain-is-precond-negative ?precond:part-of))
+    (bind ?parent-is-negative (domain-is-formula-negative ?precond:part-of))
     (return (neq (eq ?precond:type negation) ?parent-is-negative))
   )
 )
 
+(deffunction domain-is-precond-negative
+  "Mapping of the old domain-is-precond-negative to the new replacement function
+  for PDDL formulas for compatibility reasons."
+  (?precond-name)
+  (return (domain-is-formula-negative ?precond-name))
+)
+
+(deffunction domain-get-operator-for-pddl-predicate
+  (?part-of)
+
+  (do-for-fact ((?formula pddl-formula)) (eq ?formula:id ?part-of)
+    (return (domain-get-operator-for-pddl-predicate ?formula:part-of))
+  )
+  (do-for-fact ((?operator domain-operator)) (eq ?operator:name ?part-of)
+    (return ?part-of)
+  )
+  (return FALSE)
+)
+
 (defrule domain-remove-precond-on-sensed-nonval-effect-of-exog-action
-  "If an exogenous action has a precondition for a non-value predicate that is
-   also a sensed effect of the operator, then remove the precondition on the
-   effect. This means that part of the exogenous action may already have
-   occurred before the action is selected."
   (domain-operator (name ?op) (exogenous TRUE))
   (domain-predicate (name ?pred) (sensed TRUE) (value-predicate FALSE))
-  (domain-effect (part-of ?op) (predicate ?pred)
-    (param-names $?params) (param-constants $?constants))
-  ?pre <- (domain-atomic-precondition (name ?precond) (operator ?op)
-    (grounded FALSE) (predicate ?pred) (param-names $?params)
-    (param-constants $?constants))
-=>
+  (domain-effect (part-of ?op)
+                 (predicate ?pred)
+                 (param-names $?params)
+                 (param-constants $?constants))
+  ?pre <- (pddl-predicate (part-of ?precond&:(eq (domain-get-operator-for-pddl-predicate ?precond)
+                                                 ?op))
+                          (param-names $?params)
+                          (param-constants $?constants)
+                          (predicate ?pred))
+  =>
   (remove-precondition ?pre)
-  ; If there are any grounded preconditions, we need to recompute them.
   (domain-retract-grounding)
```

Comment:
Is this still necessary? If we always have the dual checks satisfied/not satisfied, then I don't think we need this, do we? (not just here, but generally)